Sources/Sentry/SentryTracer.m

```diff
@@ -515,13 +514,12 @@ - (void)captureTransactionWithProfile:(SentryTransaction *)transaction
 {
     SentryEnvelopeItem *profileEnvelopeItem =
         [SentryProfiler createProfilingEnvelopeItemForTransaction:transaction];
+
     if (!profileEnvelopeItem) {
         [_hub captureTransaction:transaction withScope:_hub.scope];
         return;
     }
 
-    stopTrackingTracerWithID(self.traceId, ^{ [SentryProfiler stop]; });
```

Comment:
This is now handled automatically from `SentryTracerConcurrency.profilerForFinishedTracer` (which is only called from `+[SentryProfiler createProfilingEnvelopeItemForTransaction:]` called on line 517 above).